src/components/LoadingSpinner.tsx

```diff
@@ -1,18 +1,50 @@
 import React from 'react';
-import { ActivityIndicator } from 'react-native';
+import { ActivityIndicator, View, StyleSheet } from 'react-native';
 
 import { colors } from '../config';
 
-import { LoadingContainer } from './LoadingContainer';
-
 type Props = {
   loading?: boolean;
 };
 
 export const LoadingSpinner = ({ loading }: Props) => {
-  return loading ? (
-    <LoadingContainer>
+  if (!loading) return null;
+
+  return (
+    <View style={styles.loadingContainer}>
       <ActivityIndicator color={colors.refreshControl} />
-    </LoadingContainer>
-  ) : null;
+    </View>
+  );
 };
+
+const styles = StyleSheet.create({
+  loadingContainer: {
+    ...StyleSheet.absoluteFillObject,
+    backgroundColor: 'rgba(255, 255, 255, 0.7)',
+    justifyContent: 'center',
+    alignItems: 'center',
+    zIndex: 10,
+  },
+});
+
+// below is the old code, needed for the app 
+// the above is temporary for the new map
+
+// import React from 'react';
+// import { ActivityIndicator } from 'react-native';
+
+// import { colors } from '../config';
+
+// import { LoadingContainer } from './LoadingContainer';
+
+// type Props = {
+//   loading?: boolean;
+// };
+
+// export const LoadingSpinner = ({ loading }: Props) => {
+//   return loading ? (
+//     <LoadingContainer>
+//       <ActivityIndicator color={colors.refreshControl} />
+//     </LoadingContainer>
+//   ) : null;
+// };
```

Comment:
so it could be used again as it is in the newest commit https://github.com/smart-village-solutions/smart-village-app-app/pull/1154/commits/7d15a3d5f2af482f630518fdc922da1732420e58#diff-4c6f3a32c46718d48a9634ef0ae21287e206c925ecbf614505b3e266a01bc6ea